lib/charms/tls_certificates_interface/v3/tls_certificates.py

```diff
@@ -0,0 +1,2002 @@
+                    "revoked": True,
+                }
+            ]
+        },
+    ],
+    "properties": {
+        "certificates": {
+            "$id": "#/properties/certificates",
+            "type": "array",
+            "items": {
+                "$id": "#/properties/certificates/items",
+                "type": "object",
+                "required": ["certificate_signing_request", "certificate", "ca", "chain"],
+                "properties": {
+                    "certificate_signing_request": {
+                        "$id": "#/properties/certificates/items/certificate_signing_request",
+                        "type": "string",
+                    },
+                    "certificate": {
+                        "$id": "#/properties/certificates/items/certificate",
+                        "type": "string",
+                    },
+                    "ca": {"$id": "#/properties/certificates/items/ca", "type": "string"},
+                    "chain": {
+                        "$id": "#/properties/certificates/items/chain",
+                        "type": "array",
+                        "items": {
+                            "type": "string",
+                            "$id": "#/properties/certificates/items/chain/items",
+                        },
+                    },
+                    "revoked": {
+                        "$id": "#/properties/certificates/items/revoked",
+                        "type": "boolean",
+                    },
+                },
+                "additionalProperties": True,
+            },
+        }
+    },
+    "required": ["certificates"],
+    "additionalProperties": True,
+}
+
+
+logger = logging.getLogger(__name__)
+
+
+class RequirerCSR:
+    """This class represents a certificate signing request from an interface Requirer."""
+
+    def __init__(
+        self, relation_id: int, application_name: str, unit_name: str, csr: str, is_ca: bool
+    ):
+        self.relation_id = relation_id
+        self.application_name = application_name
+        self.unit_name = unit_name
+        self.csr = csr
+        self.is_ca = is_ca
+
+    def __eq__(self, other):
+        return (
+            self.relation_id == other.relation_id
+            and self.application_name == other.application_name
+            and self.unit_name == other.unit_name
+            and self.csr == other.csr
+            and self.is_ca == other.is_ca
+        )
```

Comment:
```suggestion
@dataclass
class RequirerCSR:
    """This class represents a certificate signing request from an interface Requirer."""
        relation_id: int
        application_name: str
        unit_name: str
        csr: str
        is_ca: bool
```
python will give you the same `__eq__` function by default.

Reply:
Done